docs/designs/elasticdl_operator.md

```diff
@@ -0,0 +1,212 @@
+      --model_zoo 'model_zoo' --cluster_spec '' --minibatch_size '64' --log_level
+      'INFO' --dataset_fn 'dataset_fn' --loss 'loss' --optimizer 'optimizer' --callbacks
+      'callbacks' --eval_metrics_fn 'eval_metrics_fn' --custom_data_reader 'custom_data_reader'
+      --model_def 'mnist_functional_api.mnist_functional_api.custom_model' --model_params
+      '' --get_model_steps '1' --data_reader_params '' --distribution_strategy 'ParameterServerStrategy'
+      --checkpoint_steps '0' --checkpoint_dir '' --keep_checkpoint_max '0' --output
+      '' --image_name 'elasticdl:test' --job_name 'test-mnist' --master_resource_request
+      'cpu=0.2,memory=1024Mi' --master_resource_limit 'cpu=1,memory=2048Mi' --num_workers
+      '1' --worker_resource_request 'cpu=0.4,memory=1024Mi' --worker_resource_limit
+      'cpu=1,memory=2048Mi' --master_pod_priority '' --worker_pod_priority '' --num_ps_pods
+      '1' --ps_resource_request 'cpu=0.2,memory=1024Mi' --ps_resource_limit 'cpu=1,memory=2048Mi'
+      --ps_pod_priority '' --volume 'host_path=/data,mount_path=/data' --image_pull_policy
+      'Never' --restart_policy 'Never' --envs '' --extra_pypi_index 'https://pypi.org/simple'
+      --namespace 'default' --num_minibatches_per_task '2' --use_go_ps 'True' --aux_params '' --log_file_path '' --tensorboard_log_dir
+      '' --num_epochs '2' --grads_to_wait '1' --training_data '/data/mnist/train'
+      --validation_data '' --evaluation_steps '0' --evaluation_start_delay_secs '100'
+      --evaluation_throttle_secs '0' --checkpoint_dir_for_init '' --sync_version_tolerance
+      '0' --log_loss_steps '100' --use_async 'False' --lr_staleness_modulation 'False'
+    command:
+    - /bin/bash
+    env:
+    - name: MY_POD_IP
+      valueFrom:
+        fieldRef:
+          fieldPath: status.podIP
+    image: elasticdl:test
+    imagePullPolicy: Never
+    name: elasticdl-test-mnist-master
+    resources:
+      limits:
+        cpu: '1'
+        memory: 2048Mi
+      requests:
+        cpu: '0.2'
+        memory: 1024Mi
+    volumeMounts:
+    - mountPath: /data
+      name: elasticdl-test-mnist-master-volume-0
+  priorityClassName: ''
+  restartPolicy: Never
+  volumes:
+  - hostPath:
+      path: /data
+    name: elasticdl-test-mnist-master-volume-0
+```
+
+We could rewrite it as a custom ElasticDLJob object
+after the ElasticDL CRD is created.
+The following is a demo:
+
+```yaml
+apiVersion: "elasticdl.org/v1"
+kind: "ElasticDLJob"
+metadata:
+  name: "test-mnist"
+spec:
+  elasticDLArgs:
+    modelZoo: /model_zoo
+    modelDef: mnist_functional_api.mnist_functional_api.custom_model
+    trainingData: /data/mnist/train
+    validationData: /data/mnist/val
+    output: /data/output
+    dataReaderParams: ""
+    minibatchSize: 64
+    numMinibatchesPerTask: 20
+    evaluationStep: 1000
+    checkpointStep: 1000
+    checkpointDir: /data/checkpoint
+    checkpointDirForInit: ""
+    keepCheckpointMax: 0
+  elasticDLReplicaSpecs:
```

Comment:
Maybe we can remove `elasticDLReplicaSpecs `, and make `Master`, `PS`, and `Worker` to the upper level.